enterprise/models.py

```diff
@@ -386,8 +388,12 @@ def entitlements(self):
 
         """
         # Check if Enterprise Learner consents to data sharing and store the boolean result
-        learner_consent_state = self.data_sharing_consent.first()
-        learner_consent_enabled = learner_consent_state and learner_consent_state.enabled
+        DataSharingConsent = apps.get_model('consent', 'DataSharingConsent')  # pylint: disable=invalid-name
+        learner_consent_disabled = DataSharingConsent.objects.filter(
+            enterprise_customer=self.enterprise_customer,
+            username=self.username,
+            granted=False,
```

Comment:
That's right, good catch. We can do it with `granted__in=[None, False]` I believe, and I'll just add an extra test case for that.